pymc/model/transform/conditioning.py

```diff
@@ -95,7 +96,7 @@ def observe(
         for var, obs in vars_to_observations.items()
     }
 
-    valid_model_vars = set(model.free_RVs + model.deterministics)
+    valid_model_vars = set(model.free_RVs + model.deterministics + model.observed_RVs)
```

Comment:
The messagw below should be updated to not a model random variable or deterministic.

You can use model.basic_RVs in place of free + observed, and ModelValuedVar for the corresponding in the ir representation in the asser below